### DISCUSSION TITLE
Question: How to reuse the same .svg file in .vue and .liquid?

### DISCUSSION THREAD

**Discussion by albertpratomo** [uicrooks/shopify-theme-lab#139] [Q&A] [answered]
Hello. I need to render a svg component in liquid and vue. By default liquid has these icon snippets, but I think it's not possible to render liquid snippets within Vue components, right?

So Im thinking to load .svg in both liquid snippets and vue components. Do you have suggestions about this?

I know I can use renderless-component and render the icon via liquid. But in this case I need to use it as full vue component because the component is quite complex and dynamic. Much thanks before 🙏🏼 .

**Comment by sergejcodes** [comment, marked as answer]
I think the only way, without having duplicates, is to refactor the icons to vue components. That way they can be loaded in other vue files as well as in the liquid templates.

**Reply by albertpratomo** [reply]
I see. That's a valid solution. Only downside is the icon will be JS rendered, but I think it's fine..

**Comment by thevrus** [comment]
You can use SVG sprites
https://css-tricks.com/svg-sprites-use-better-icon-fonts/
